test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -253,3 +235,127 @@ func schedulePods(config *testConfig) int32 {
 		time.Sleep(1 * time.Second)
 	}
 }
+
+func (nodeAffinity *nodeAffinity) mutateNode(numNodes int) []testutils.CountToStrategy {
+	numGroups := nodeAffinity.numGroups
+	nodeAffinityKey := nodeAffinity.nodeAffinityKey
+	nodeStrategies := make([]testutils.CountToStrategy, 0, 10)
+	for i := 0; i < numGroups; i++ {
+		nodeStrategies = append(nodeStrategies, testutils.CountToStrategy{
+			Count:    numNodes / numGroups,
+			Strategy: testutils.NewLabelNodePrepareStrategy(nodeAffinityKey, fmt.Sprintf("%v", i)),
+		})
+	}
+	return nodeStrategies
+}
+
+func (nodeAffinity *nodeAffinity) mutatePod(numPods int, pod *v1.Pod) []*v1.Pod {
+	numGroups := nodeAffinity.numGroups
+	nodeAffinityKey := nodeAffinity.nodeAffinityKey
+	podList := make([]*v1.Pod, 0, numGroups)
+	for i := 0; i < numGroups; i++ {
+		pod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "sched-perf-node-affinity-pod-",
+			},
+			Spec: testutils.MakePodSpec(),
+		}
+		pod.Spec.Affinity = &v1.Affinity{
+			NodeAffinity: &v1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      nodeAffinityKey,
+									Operator: nodeAffinity.Operator,
+									Values:   []string{fmt.Sprintf("%v", i)},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		podList = append(podList, pod)
+	}
+	return podList
+}
+
+// Interface that every predicate or priority structure should implement.
+type UpdateNodePodConfig interface {
+	// mutateNode mutates the node strategies
+	mutateNode(int)
+	// mutatePod mutates the pod configs
+	mutatePod(int, *v1.Pod)
+}
+
+
+// TODO: As of now, returning configs hardcoded, need to read from yaml or some other file.
+func readInConfiguration() *schedulerPerfConfig {
+	return &schedulerPerfConfig{
+		NodeAffinity: nodeAffinity{
+			Enabled:         false,
+			numGroups:       10,
+			nodeAffinityKey: "kubernetes.io/sched-perf-node-affinity",
+			Operator:        v1.NodeSelectorOpIn,
+		},
+
+		InterpodAffinity: interpodAffinity{
+			Enabled:     false,
+			Operator:    metav1.LabelSelectorOpIn,
+			affinityKey: "security",
+			Labels:      map[string]string{"security": "S1"},
+			TopologyKey: "region",
+		},
+	}
+}
+
+func (inputConfig *schedulerPerfConfig) mutate(config *testConfig) {
```

Comment:
1) don't we need support for taints, serviceAffinity, and so on in this function ?
2) also, make sure you use interfaces wherever possible... the comment above about how the interface definition didn't need to conform scares me a little bit that we're overcoupling in this function and not leveraging the generic interface properties.